### ISSUE TITLE
Removed feature namespace URI from the config parameters and test namespace persistent in the adapter

### ISSUE THREAD

**Issue by rwgdrummer** [ngageoint/geowave#668]
(1) The Feature Namespace URI was used to force a namespace for all feature type loaded through GeoServer.  The better alternative is to support changing the namespace during ingest.  The only remaining concern is construction of the feature as part of defining a layer in the UI.   GeoServer does not support changing the name space in this case.  However, the forced namespace for all features using a data store is too broad of a solution.

(2) Add a unit test for namespace preservation in the feature data adapter.  It does work. Just need to continuously check it.

**Comment by datasedai** [comment]
GeoWave feature types are namespace aware.  The namespace is encoded via FeatureDataAdapter and persisted within the Adapter Store.  To deal with GeoServer, GeoWave has code within its implementation of ContentDataSource to null out namespaces in qualified names of feature types it reports to GeoServer (createTypeNames function).  Likewise, it transforms all requests for features by nulling out the namespaces (getFeatureSource functions).  

Before the modification mentioned in this issue, GeoWave advertised and accepted the "namespace" data store parameter (inherited from GeoServer Workspace), which was then used by SimpleFeatureTypeBuilder in FeatureDataAdapter to encode the namespace in generated SimpleFeature objects.  This value overrode the persisted namespace, ensuring features were generated with the Geoserver Workspace namespace.

By removing usage of the "namespace" parameter within GeoServer UI panel, the FeatureDataAdapter is forced to use the persisted namespace, which is dependent on how the format plugin decides to set the value when ingested.  If not specified or explicitly nulled out, SimpleFeatureTypeBuilder will use the default value of "http://www.opengis.org/gml".

When exporting features to HTML, GeoServer will retrieve the features internally, and then access the feature layer to retrieve template information for styling the feature.  Because the features are qualified with a namespace that is different than the GeoServer workspace they're being served from, the internal code cannot find the feature layer, and it throws the following exception:

`java.lang.IllegalArgumentException: Can't find neither a FeatureType nor a CoverageInfo or WMSLayerInfo named <namespace>:<layer>`

I think we should respect the Geoserver workspace namespace URI, and continue to allow the Geoserver "namespace" data store parameter to dictate the namespace we use to generate simple features for WFS.

I say this for two reasons:

1. Geowave is a Database, and as stated in ContentDataSource, Databases use unqualified names: 
> "An example of a datastore that might use qualified names is WFS, where in each entry corresponds to a WFS "Feature Type", which have namespace qualified name. Other datastores (such as databases) use unqualified names."

2. The namespace makes up the qualified name of the feature layer.  Suppose I need to ingest the same features into two different workspaces.  If I allow the ingested namespace to be forced to the original ingested namespace, then I will have no way to distinguish between content retrieved from both layers, as they will report the same qualified name.  I feel as though any data that lives within a certain geoserver workspace should be qualified as having originated from that workspace to retain data provenance.  It just makes sense.

To check, I reviewed JDBCDataStore, and they expose unqualified types as well.  They pass namespaces specified via ContentEntry createFeatureSource directly to the SimpleFeatureType used to generate those features, and if it is not specified (or null), then they use the namespace specified via the "namespace" parameter directly.

I think this issue is made more complicated than it needs to be due to the fact that we have conflicting desires:

1. We want to retain the capability in Geowave to be namespace aware.  When we ingest data, we want to be able to retain the original ingested namespace.  This is stored in the SimpleFeatureType that is persisted with the Geowave FeatureDataAdapter.

2. We want to serve features as a Geoserver ContentDataSource, which encourages us to use the namespace specified via the Geoserver workspace, and to expose our own types as unqualified.

As a solution, I recommend we continue to use the method we originally were using, by advertising the "namespace" parameter to Geoserver.

**Comment by rwgdrummer** [comment]
My understanding is that we cannot easily support both objectives without adding addition code.
We identified a fix to use the feature namespace in GeoWaveGTDataStore
@Override
	protected List<Name> createTypeNames()
			throws IOException {
		final List<Name> names = new ArrayList<>();
		final CloseableIterator<DataAdapter<?>> adapters = adapterStore.getAdapters();
		while (adapters.hasNext()) {
			final DataAdapter<?> adapter = adapters.next();
			if (adapter instanceof GeotoolsFeatureDataAdapter) {
				names.add(((GeotoolsFeatureDataAdapter) adapter).getType().getName());
			}
		}
		adapters.close();
		return names;
	}

We have null check in place to not over-ride this name if the data store namespace parameter is null.  However, according to what is stated, the namespace parameter is never null if provided because GeoServer defaults the namespace to the workspace namespace.  So, our best bet is to follow #2 and add the few lines of code back to capture the name space in the plugin parameters.  It is still a good idea to apply the other fix.

**Comment by rfecher** [comment]
It seems the contract for being a ContentDataStore is to serve the features using its namespaceUri so I don't think we have a choice.  I think we need to override with the geoserver workspace namespace URI based on @datasedai research.  While it may or may not be clear whether the underlying JDBC data store should even have qualified type names that become overridden in the ContentDataStore, I think a shapefile inherently could have an underlying qualified type name much the same as geowave.  So I'd say considering the ShapefileDataStore explicitly overrides the underlying namespace with the ContentDataStore namespaceUri that presumably comes from the geoserver workspace, we should do the same: https://github.com/geotools/geotools/blob/master/modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileDataStore.java#L140

Basically, I think that line pretty clearly supports @datasedai's point - @rwgdrummer do you see any issues with simply always overriding the underlying FeatureType name with the namespaceUri of the workspace essentially similar to the ShapefileDataStore?

**Comment by datasedai** [comment]
Ok, I'm going to split the difference.

I will make it so it uses the GeoServer namespace by default, if it is set.  If it is not set (someone manually removes it from datastore.xml), then it will use the feature namespace.  So, I will fix the bug in createTypeNames(), but modify the code to still advertise and use namespace first and foremost.
